File: packages/room-server/src/user/entities/user.entity.ts
import { IdWorker } from 'shared/helpers/snowflake';
import { BeforeInsert, Column, Entity, PrimaryColumn } from 'typeorm';

/**
 * Workbench-User
 */
@Entity('user')
export class UserEntity {
  @Column({
    name: 'uuid',
    nullable: true,
    unique: true,
    comment: 'user uuid',
    length: 32,
  })
  uuid!: string;

  @Column({
    name: 'nick_name',
    nullable: true,
    comment: 'nick name',
    length: 50,
  })
  nikeName?: string;

  @Column({
    name: 'mobile_phone',
    nullable: true,
    comment: 'mobile phone number',
    length: 50,
  })
  mobilePhone?: string;

  @Column({
    name: 'email',
    nullable: true,
    comment: 'email',
    length: 100,
  })
  email?: string;

  @Column({
    name: 'password',
    nullable: true,
    comment: 'password',
    length: 255,
  })
  password?: string;

  @Column({
    name: 'avatar',
    nullable: true,
    comment: 'avatar',
    length: 255,
  })
  avatar?: string;

  @Column({
    name: 'color',
    nullable: true,
    comment: 'default avatar color',
    width: 10,
    type: 'int',
  })
  color?: number;

  @Column({
    name: 'gender',
    comment: 'gender',
    length: 1,
    default: '1',
  })
  gender!: string;

  @Column({
    name: 'remark',
    nullable: true,
    comment: 'remark',
    length: 255,
  })
  remark?: string;

  @Column({
    name: 'ding_open_id',
    nullable: true,
    comment: 'The unique ID in the open application',
    length: 255,
  })
  dingOpenId?: string;

  @Column({
    name: 'ding_union_id',
    nullable: true,
    comment: 'The unique ID in the Dingding developer enterprise',
    length: 255,
  })
  dingUnionId?: string;

  @Column('timestamp', {
    name: 'last_login_time',
    nullable: true,
    comment: 'updated time',
  })
  lastLoginTime?: Date;

  @Column({
    name: 'locale',
    nullable: true,
    comment: 'locale',
    length: 50,
  })
  locale?: string;

  @Column({
    name: 'is_social_name_modified',
    nullable: true,
    comment: 'have you modified the nickname as a third -party IM user? 0: No; 1: Yes; 2: Not third -party users of IM',
    width: 1,
    type: 'tinyint',
  })
  isSocialNameModified?: number;

  // picked from base entity
  @PrimaryColumn('bigint')
  id!: string;

  @Column({
    name: 'is_deleted',
    comment: 'deleted mark (0: No, 1: Yes)',
    unsigned: true,
    default: () => false,
  })
  isDeleted!: boolean;

  @Column('timestamp', {
    name: 'created_at',
    comment: 'created time',
    default: () => 'CURRENT_TIMESTAMP',
  })
  createdAt!: Date;

  @Column('timestamp', {
    name: 'updated_at',
    nullable: true,
    comment: 'updated time',
    default: () => 'CURRENT_TIMESTAMP',
  })
  updatedAt?: Date;

  @BeforeInsert()
  beforeInsert() {
    this.id = IdWorker.nextId().toString();
  }
}

File: backend-server/application/src/main/java/com/apitable/user/entity/UserEntity.java


package com.apitable.user.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import java.io.Serializable;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

/**
 * <p>
 * User Table.
 * </p>
 *
 * @author Mybatis Generator Tool
 */
@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode
@TableName(keepGlobalPrefix = true, value = "user")
public class UserEntity implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * Primary Key.
     */
    @TableId(value = "id", type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * User ID.
     */
    private String uuid;

    /**
     * Nick Name.
     */
    private String nickName;

    /**
     * code.
     */
    private String code;

    /**
     * Phone Number.
     */
    private String mobilePhone;

    /**
     * Email.
     */
    private String email;

    /**
     * Password.
     */
    private String password;

    /**
     * Avatar.
     */
    private String avatar;

    /**
     * default avatar color number.
     */
    private Integer color;

    /**
     * Time Zone.
     */
    private String timeZone;

    /**
     * Gender.
     */
    private String gender;

    /**
     * Remark.
     */
    private String remark;

    /**
     * Language.
     */
    private String locale;

    /**
     * Unique identification in open application of DingTalk.
     */
    private String dingOpenId;

    /**
     * Unique identifier in the DingTalk developer enterprise.
     */
    private String dingUnionId;

    /**
     * Last Login Time.
     */
    private LocalDateTime lastLoginTime;

    /**
     * Whether the nickname has been modified as a third-party IM user. 0: No;
     * 1: Yes; 2: Not an IM third-party user.
     */
    private Integer isSocialNameModified;

    /**
     * Whether to cancel the cool off period (1: Yes, 0: No).
     */
    private Boolean isPaused;

    /**
     * Delete Tag (1: Yes, 0: No).
     */
    @TableLogic
    private Boolean isDeleted;

    /**
     * Create Time.
     */
    private LocalDateTime createdAt;

    /**
     * Update Time.
     */
    private LocalDateTime updatedAt;


}
